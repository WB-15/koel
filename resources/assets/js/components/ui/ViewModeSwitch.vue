<template>
  <span class="view-modes">
    <label
      :class="{ active: value === 'thumbnails' }"
      class="thumbnails"
      data-testid="view-mode-thumbnail"
      title="View as thumbnails"
    >
      <input v-model="value" class="hidden" name="view-mode" type="radio" value="thumbnails">
      <i class="fa fa-th-large"/>
      <span class="hidden">View as thumbnails</span>
    </label>

    <label
      :class="{ active: value === 'list' }"
      class="list"
      data-testid="view-mode-list"
      title="View as list"
    >
      <input v-model="value" class="hidden" name="view-mode" type="radio" value="list">
      <i class="fa fa-list"/>
      <span class="hidden">View as list</span>
    </label>
  </span>
</template>

<script lang="ts" setup>
import { computed } from 'vue'

const props = withDefaults(defineProps<{ modelValue?: ArtistAlbumViewMode }>(), { modelValue: 'thumbnails' })
const emit = defineEmits(['update:modelValue'])

const value = computed({
  get: () => props.modelValue,
  set: value => emit('update:modelValue', value)
})
</script>

<style lang="scss" scoped>
.view-modes {
  display: flex;
  width: 64px;
  border: 1px solid rgba(255, 255, 255, .2);
  border-radius: 5px;
  overflow: hidden;

  label {
    width: 50%;
    text-align: center;
    line-height: 2rem;
    font-size: 1rem;
    margin-bottom: 0;
    cursor: pointer;

    &.active {
      background: var(--color-text-primary);
      color: var(--color-bg-primary);
    }
  }

  @media only screen and(max-width: 768px) {
    margin-top: 8px;
  }
}
</style>
