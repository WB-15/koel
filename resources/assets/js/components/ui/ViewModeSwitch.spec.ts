import { expect, it } from 'vitest'
import ComponentTestCase from '@/__tests__/ComponentTestCase'
import ViewModeSwitch from './ViewModeSwitch.vue'

new class extends ComponentTestCase {
  protected test () {
    it.each<[ArtistAlbumViewMode]>([['thumbnails'], ['list']])('renders %s mode', mode => {
      expect(this.render(ViewModeSwitch, {
        props: {
          mode
        }
      }).html()).toMatchSnapshot()
    })
  }
}
