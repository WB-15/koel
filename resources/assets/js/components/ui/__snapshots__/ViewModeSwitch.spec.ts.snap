// Vitest Snapshot v1

exports[`renders list mode 1`] = `<span class="view-modes" data-v-0c4113b0=""><label class="thumbnails" data-testid="view-mode-thumbnail" title="View as thumbnails" data-v-0c4113b0=""><input class="hidden" name="view-mode" type="radio" value="thumbnails" data-v-0c4113b0=""><i class="fa fa-th-large" data-v-0c4113b0=""></i><span class="hidden" data-v-0c4113b0="">View as thumbnails</span></label><label class="active list" data-testid="view-mode-list" title="View as list" data-v-0c4113b0=""><input class="hidden" name="view-mode" type="radio" value="list" data-v-0c4113b0=""><i class="fa fa-list" data-v-0c4113b0=""></i><span class="hidden" data-v-0c4113b0="">View as list</span></label></span>`;

exports[`renders thumbnails mode 1`] = `<span class="view-modes" data-v-0c4113b0=""><label class="active thumbnails" data-testid="view-mode-thumbnail" title="View as thumbnails" data-v-0c4113b0=""><input class="hidden" name="view-mode" type="radio" value="thumbnails" data-v-0c4113b0=""><i class="fa fa-th-large" data-v-0c4113b0=""></i><span class="hidden" data-v-0c4113b0="">View as thumbnails</span></label><label class="list" data-testid="view-mode-list" title="View as list" data-v-0c4113b0=""><input class="hidden" name="view-mode" type="radio" value="list" data-v-0c4113b0=""><i class="fa fa-list" data-v-0c4113b0=""></i><span class="hidden" data-v-0c4113b0="">View as list</span></label></span>`;
